scripts/release/orthoinference/deprecated/wrapper_ortho_inference.pl

```diff
@@ -0,0 +1,144 @@
+    'release_date:s' => \$release_date,
+    'debug' => \$debug
+);
+$source_db || die "Need source database: -source_db [database from which to copy content]\n";
+$release || die "Need release version: -release [Reactome version number]\n";
+$release_date || die "Need release date: -release_date [scheduled release date as yyyy-mm-dd]\n";
+$user ||= $GKB::Config::GK_DB_USER;
+$pass ||= $GKB::Config::GK_DB_PASS;
+$host ||= $GKB::Config::GK_DB_HOST;
+$port ||= $GKB::Config::GK_DB_PORT;
+$target_db ||= 'release_current';
+$source_species ||= 'hsap'; # Human is defined as default source species
+
+my %db_options = (
+    '-host' => $host,
+    '-port' => $port,
+    '-user' => $user,
+    '-pass' => $pass,
+);
+my $db_option_string = create_db_option_string(\%db_options);
+
+#create database handle - needed to create the source database copy which will be used to add orthology data
+my $dbc = DBI->connect(
+    "DBI:mysql:database=$source_db;host=$host;port=$port",
+    $user,
+    $pass,
+    { RaiseError => 1, AutoCommit => 1}
+);
+if (!$dbc) {
+    $logger->error_die("Error connecting to database; $DBI::errstr\n");
+}
+
+# Create and populate target db from source db
+system("mysql -u$user -p$pass -e 'drop database if exists $target_db; create database $target_db'") == 0 or die "$?";
+run("mysqldump --opt -u$user -p$pass -h$host $source_db | mysql -u$user -p$pass -h$host $target_db") == 0 or die "$?";
+
+# Some datamodel changes are required for running the script, mainly to adjust defining attributes in order to avoid
+# either duplication or merging of instances in the inference procedure, plus introduction of some additional attributes
+my $exit_value = run("perl tweak_datamodel.pl -db $target_db $db_option_string");
+if ($exit_value != 0) {
+    $logger->error_die("problem encountered during tweak_datamodel, aborting\n");
+}
+
+#run script for each species (order defined in config.pm)
+my @species_to_exclude = qw/mtub/;
+foreach my $current_species (@species) {
+    $logger->info("wrapper_ortho_inference: considering species=$current_species\n");
+    next if $current_species eq $source_species; #skip source species in species list
+    next if any { $_ eq $current_species } @species_to_exclude;
+    if ($target_species) {
+       next unless $current_species eq $target_species;
+    }
+    $logger->info("wrapper_ortho_inference: running infer_events script\n");
+    run(
+        "perl infer_events.pl -db $target_db -r $release -from $source_species -sp $target_species " .
+        " -release_date $release_date -thr 75 @ARGV $db_option_string"
+    ); #run script with 75% complex threshold
+}
+`chgrp reactome $release/* 2> /dev/null`; # Allows all group members to read/write compara release files
+
+$logger->info("wrapper_ortho_inference: run clean up scripts\n");
+#These are two "clean-up" scripts to remove unused PhysicalEntities and to update display names
+run("perl remove_unused_PE.pl -db $target_db $db_option_string");

```

Comment:
I thought the functionality of `remove_unused_PE` was either deprecated, or moved into the Java implementation by now... Did I miss something?

Reply:
oh right... this script is now deprecated.